umbra/controller.py

```diff
@@ -230,6 +231,37 @@ def on_response(chrome_msg):
                 publish(payload, exchange=self._exchange, routing_key=client_id)
 
         def post_outlinks(outlinks=None):
+            def prune_outlinks(dirty_links, block_list=None):
+                '''
+                Remove URL fragments, javascript URLs, and any other designated URLs from the list.
+                '''
+                links = set()
+                dirty_links = set(dirty_links)
+
+                self.logger.info('Pruning link fragements...')
+                for link in dirty_links:
+                    # remove link fragments
+                    link = urlcanon.parse_url(link)
+                    urlcanon.canon.remove_fragment(link)
+                    link = str(link).strip()
+
+                    # we can't crawl javascript URLs
+                    if not link.startswith('javascript'):
```

Comment:
Instead of `not link.startswith('javascript')` I would check `link.scheme != b'javascript'` (before re-stringifying `link`). 

Or perhaps a whitelist of schemes would be preferable here? I could imagine `data:` urls or other junk getting in. Maybe something like this?
```
if link.scheme in (b'http', b'https', b'ftp'):
    links.add(link)
```

Reply:
I like the scheme whitelist approach. I'll make the change in QA and test.